src/js/content/checkWorkerEndpointCSP.ts

```diff
@@ -18,13 +18,13 @@ export function checkWorkerEndpointCSP(
   origin: string,
 ): boolean {
   const host = ORIGIN_HOST[origin];
-  const cspHeaders = getCSPHeadersFromWebRequestResponse(response).map(
-    h => h.value,
-  );
-  const cspReportHeaders = getCSPHeadersFromWebRequestResponse(
-    response,
-    true,
-  ).map(h => h.value);
+  const cspHeaders = getCSPHeadersFromWebRequestResponse(response)
+    .map(h => h.value)
+    .filter((header): header is string => !!header);
```

Comment:
Hmm what does the return type annotation here do? 🤔

Reply:
For esoteric reasons (longstanding bug?) filtering an array like this does not inherently narrow the type. I.e. we're trying to go from `Array<string | undefined` to `Array<string>`, but for whatever reason TypeScript doesn't understand that we're filtering out all the the `undefined`.

If, however, we use explicit [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) here, it does work. I haven't found a good reason why. We **are** giving up some safety here, since this is a form of casting.

I realized that another alternative would be to use `reduce` instead of `map`/`filter`, since I believe TS understands that correctly. Maybe that would be better?

Reply:
Eh, I think leaving this as filter + guard is better than reduce:

```js
  const cspHeaders = getCSPHeadersFromWebRequestResponse(response)
    .reduce((acc: string[], h) => {
      if (h.value !== undefined) {
        acc.push(h.value)
      }
      return acc;
    }, []);
```